pkg/sip/media_port.go

```diff
@@ -184,30 +235,104 @@ func (p *MediaPort) SetOffer(offerData []byte) (*sdp.Answer, *MediaConf, error)
 }
 
 func (p *MediaPort) SetConfig(c *MediaConf) error {
+	var crypto string
+	if c.Crypto != nil {
+		crypto = c.Crypto.Profile.String()
+	}
 	p.log.Infow("using codecs",
 		"audio-codec", c.Audio.Codec.Info().SDPName, "audio-rtp", c.Audio.Type,
 		"dtmf-rtp", c.Audio.DTMFType,
+		"srtp", crypto,
 	)
 
+	p.port.SetDst(c.Remote)
+	var (
+		sess rtp.Session
+		err  error
+	)
+	if c.Crypto != nil {
+		sess, err = srtp.NewSession(p.port, c.Crypto)
+	} else {
+		sess = rtp.NewSession(p.port)
+	}
+	if err != nil {
+		return err
+	}
+
 	p.mu.Lock()
 	defer p.mu.Unlock()
-	if ip := c.Remote; ip.IsValid() {
-		p.conn.SetDestAddr(&net.UDPAddr{
-			IP:   ip.Addr().AsSlice(),
-			Port: int(ip.Port()),
-		})
-	}
+	p.port.SetDst(c.Remote)
 	p.conf = c
+	p.sess = sess
 
-	p.setupOutput()
+	if err = p.setupOutput(); err != nil {
+		return err
+	}
 	p.setupInput()
 	return nil
 }
 
+func (p *MediaPort) rtpLoop(sess rtp.Session) {
+	// Need a loop to process all incoming packets.
+	first := true
+	for {
+		r, ssrc, err := sess.AcceptStream()
+		if err != nil {
+			if !errors.Is(err, net.ErrClosed) {
+				p.log.Errorw("cannot accept RTP stream", err)
+			}
+			return
+		}
+		p.mediaReceived.Break()
+		if first {
+			first = false
+			p.log.Infow("accepting media", "ssrc", ssrc)
+			go p.rtpReadLoop(r)
+		} else {
+			p.log.Warnw("ignoring media", nil, "ssrc", ssrc)
+		}
+	}
+}
+
+func (p *MediaPort) rtpReadLoop(r rtp.ReadStream) {
+	buf := make([]byte, 1500)
```

Comment:
Can we also throw an error here if but overflow?

Reply:
Done, same as above.